### ISSUE TITLE
tflite to onnx    nhwc is weird

### ISSUE THREAD

**Issue by kspark1990** [onnx/tensorflow-onnx#1424]
my model   tflite input shape is 1,224,224,3  nhwc
but converted model is *,224,3,224 (nhwc)  in unity barracuda
If you know how to fix it, please teach me

**Comment by TomWildenhain-Microsoft** [comment]
You can tell tf2onnx the input shape with the `--inputs` flag.  `--inputs input_name[1,224,224,3]` should fix the issue.

**Comment by guschmue** [comment]
*,224,3,224 is wired - that is neither NCHW nor NHWC. If the tflite model has 1,224,224,3 I don't know how we'd loose that - something fishy going on. Is this some public mobilenet?

**Comment by TomWildenhain-Microsoft** [comment]
Oh, yeah, I misread it as just losing the first dim.  @kspark1990 are you able to upload the model you are converting?

**Comment by kspark1990** [comment]
@TomWildenhain-Microsoft  I used model    https://teachablemachine.withgoogle.com/    Image preject

**Comment by TomWildenhain-Microsoft** [comment]
I think teachable machine picks a pretrained model based on the input data.  To make sure we get the right one, @kspark1990 can you upload a model that has this issue?

**Comment by kspark1990** [comment]
this model [https://drive.google.com/drive/folders/1bC_ZrgiEddt_prYm-dQ4yDCSdJCZ-Riz?usp=sharing](url)

**Comment by TomWildenhain-Microsoft** [comment]
I tried converting the model and the shapes were fine for me.  What flags did you use?
My results:
https://www.dropbox.com/sh/hd340jcv1ng7vsq/AACnTIAHX09zMjhl0veAjtNXa?dl=0

Did you try viewing the converted models in Netron?
https://netron.app/

**Comment by kspark1990** [comment]
I used just 'python -m tf2onnx.convert --saved-model tensorflow-model-path --output model.onnx' 
would please teach me how to convert that your results
I didn't understand everything 'README.md' because my English wasn't good enough.

**Comment by TomWildenhain-Microsoft** [comment]
I converted the tflite model with:

`python -m tf2onnx.convert --tflite path/to/model.tflite --output desired/path/output.onnx`

It sounds like you converted the saved model (not tflite)

**Comment by TomWildenhain-Microsoft** [comment]
I converted the saved_model with `python -m tf2onnx.convert --saved-model C:\Users\tomwi\Downloads\converted_savedmodel\model.savedmodel --opset 13 --output C:\Users\tomwi\Downloads\converted_savedmodel\model.onnx` and also got a correct result with shape *,224,224,3

![image](https://user-images.githubusercontent.com/67606533/113191076-1fd16e00-922b-11eb-920b-a2ca3972ddb6.png)

**Comment by kspark1990** [comment]
I try to convert your reply but it is correct in https://netron.app/  but it is not correct in unity
<img width="308" alt="onnx netron app" src="https://user-images.githubusercontent.com/32832080/113228821-b99f2680-92d0-11eb-854d-67a1da7f4ce3.PNG">

<img width="242" alt="onnx unity" src="https://user-images.githubusercontent.com/32832080/113228872-d0de1400-92d0-11eb-87d6-99053d4489f4.png">

**Comment by TomWildenhain-Microsoft** [comment]
Can you try adding the `--inputs-as-nchw sequential_3_input:0` flag?

**Comment by kspark1990** [comment]
--inputs-as-nchw sequential_3_input:0   it works!!

**Comment by TomWildenhain-Microsoft** [comment]
Nice!

**Comment by DeveloperNo579212** [comment]
Inspired by this: Using teachablemachine I created a the simplest image project, three classes, "10 images lefthand", "10 images righthand", "10 images no hand".  I validate the model on teachablemachine using training image with expected 100% match.  I export model to saved_model, convert using tf2onnx to onnx import in ML.net and do validation, but not accurate and not 100% at all based on same input.  Using jpg as input, using png as input same result?  

Did not find option "--inputs-as-nchw" making any difference and no clue on how to proceed.
Onnx seems ok on dimensions in netron.  Easy to produce I guess, using latest versions of tf2onnx, tried different opsets etc.
